exchange/src/client_manager/client_manager.cpp

```diff
@@ -3,58 +3,67 @@
 namespace nutc {
 namespace manager {
 
-inline bool
-ClientManager::user_exists(const std::string& uid) const
+bool
+ClientManager::user_exists(const std::string& id) const
 {
-    return clients.find(uid) != clients.end();
+    return clients.find(id) != clients.end();
 }
 
-inline bool
-ClientManager::user_holds_stock(const std::string& uid, const std::string& ticker) const
+void
+ClientManager::set_client_pid(const std::string& id, pid_t pid)
+{
+    if (!user_exists(id))
+        return;
+
+    clients[id].pid = pid;
+}
+
+bool
+ClientManager::user_holds_stock(const std::string& id, const std::string& ticker) const
 {
-    if (!user_exists(uid)) [[unlikely]]
+    if (!user_exists(id)) [[unlikely]]
         return false;
 
-    return clients.at(uid).holdings.find(ticker) != clients.at(uid).holdings.end();
+    return clients.at(id).holdings.find(ticker) != clients.at(id).holdings.end();
```

Comment:
```suggestion
    return clients[id].holdings.contains(ticker);
```

also should go into the header

Reply:
No, this is a const function and clients[id] modifies the member variable clients